src/lib/y2packager/resolvable.rb

```diff
@@ -0,0 +1,133 @@
+  #
+  # This class represents a libzypp resolvable object (package, pattern, patch,
+  # product, source package)
+  #
+  # @note The returned Resolvables might not be valid anymore after changing
+  #   the package manager status (installing/removing packages, changing
+  #   repositories, etc.). After such a change you need to load the resolvables
+  #   again, avoid storing them for later if possible.
+  #
+  # @example All installed packages
+  #   Y2Packager::Resolvable.find(kind: :package, status: :installed)
+  #
+  # @example Available (not installed) "yast2" packages
+  #   Y2Packager::Resolvable.find(kind: :package, status: :available, name: "yast2")
+  #
+  # @example Lazy loading
+  #   res = Y2Packager::Resolvable.find(kind: :package, status: :installed)
+  #   # the `summary` attribute is loaded from libzypp when needed
+  #   res.each {|r| puts "#{r.name} - {r.summary}"}
+  #
+  # @example Preloading the attributes
+  #   # the `summary` attribute is loaded from libzypp already at the initial state
+  #   res = Y2Packager::Resolvable.find(kind: :package, status: :installed, [:summary])
+  #   # this returns the cached `summary` attribute, this is much more efficient
+  #   res.each {|r| puts "#{r.name} - {r.summary}"}
+  #
+  # @since 4.2.6
+  class Resolvable
+    include Yast::Logger
+
+    #
+    # Find the resolvables which match the input parameters. See Yast::Pkg.Resolvables
+    #
+    # @param params [Hash<Symbol,Object>] The search filter, only the matching resolvables
+    #    are returned.
+    # @param preload [Array<Symbol>] The list of attributes which should be preloaded.
+    #    The missing attributes are lazy loaded, however for performance reasons
+    #    you might ask to preload the attributes right at the beginning and avoid
+    #    querying libzypp again later.
+    # @return [Array<Y2Packager::Resolvable>] Found resolvables or empty array if nothing found
+    # @see https://yast-pkg-bindings.surge.sh/ Yast::Pkg.Resolvables
+    def self.find(params, preload = [])
+      attrs = (preload + UNIQUE_ATTRIBUTES).uniq
+      Yast::Pkg.Resolvables(params, attrs).map { |r| new(r) }
+    end
+
+    #
+    # Is there any resolvable matching the requested parameters? This is similar to
+    # the .find method, just instead of a resolvable list it returns a simple Boolean.
+    #
+    # @param params [Hash<Symbol,Object>] The requested attributes
+    # @return [Boolean] `true` if any matching resolvable is found, `false` otherwise.
+    # @see .find
+    def self.any?(params)
+      Yast::Pkg.AnyResolvable(params)
+    end
+
+    #
+    # Constructor, initialize the object from a pkg-bindings resolvable hash.
+    #
+    # @param hash [Hash<Symbol,Object>] A pkg-bindings resolvable hash.
+    def initialize(hash)
+      from_hash(hash)
+    end
+
+    #
+    # Dynamically load the missing attributes from libzypp.
+    #
+    # @param method [Symbol] the method called
+    # @param _args no used so far
```

Comment:
maybe raise NOMETHOD error imediatelly if args is passed? It basically makes `a.version` identical to `a.version(force: true, blabla: "Yes")`